### PR TITLE
[Storage][Sharding] Add a LedgerDb struct, to prepare split ledger db to multiple RocksDB instances.

### PR DESCRIPTION
**Pull request by grao1991**

### CODE REVIEW

File: config/src/config/storage_config.rs [hunk 1]
```diff
@@ -65,9 +65,13 @@ pub struct RocksdbConfigs {
     pub ledger_db_config: RocksdbConfig,
     pub state_merkle_db_config: RocksdbConfig,
     // Note: Not ready for production use yet.
+    // TODO(grao): Deprecate this flag and use the split_ledger_db_to_individual_dbs below.
     pub use_state_kv_db: bool,
     // Note: Not ready for production use yet.
     pub use_sharded_state_merkle_db: bool,
+    // Note: Not ready for production use yet.
+    // TODO(grao): Add RocksdbConfig for individual DBs when necessary.
+    pub split_ledger_db_to_individual_dbs: bool,
```
Comment (msmouse) [inline config/src/config/storage_config.rs:74]: nit: is the name better than something short, for example just `split_ledger_db`?

File: storage/aptosdb/src/db_options.rs [hunk 2]
```diff
@@ -33,6 +33,46 @@ pub(super) fn ledger_db_column_families() -> Vec<ColumnFamilyName> {
...
+        /* empty cf */ DEFAULT_COLUMN_FAMILY_NAME,
+        DB_METADATA_CF_NAME,
+        WRITE_SET_CF_NAME,
+    ]
+}
+
+pub(super) fn ledger_metadata_db_column_families() -> Vec<ColumnFamilyName> {
+    vec![
+        /* empty cf */ DEFAULT_COLUMN_FAMILY_NAME,
+        DB_METADATA_CF_NAME,
```
Comment (areshand) [inline storage/aptosdb/src/db_options.rs:68]: why does this CF exists in write_set_db_column_families as well?

File: storage/aptosdb/src/db_options.rs [hunk 3]
```diff
@@ -33,6 +33,46 @@ pub(super) fn ledger_db_column_families() -> Vec<ColumnFamilyName> {
...
+        EPOCH_BY_VERSION_CF_NAME,
+        LEDGER_INFO_CF_NAME,
+        TRANSACTION_INFO_CF_NAME,
```
Comment (areshand) [inline storage/aptosdb/src/db_options.rs:71]: what is the criteria for having this in this CF not the txn CF?

File: storage/aptosdb/src/ledger_db.rs [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+            &DbMetadataValue::Version(version),
+        )
+    }
+
+    pub(crate) fn create_checkpoint(
+        _db_root_path: impl AsRef<Path>,
+        _cp_root_path: impl AsRef<Path>,
+    ) -> Result<()> {
+        // TODO(grao): Implement this function.
+        Ok(())
```
Comment (msmouse) [inline storage/aptosdb/src/ledger_db.rs:137]: nit: todo!() is for cases like this no?

File: storage/aptosdb/src/db_options.rs (hunk 2 above)
Comment (grao1991) [reply storage/aptosdb/src/db_options.rs:68]: Every db keeps its own metadata, to record some information like commit progress and pruner progress.

File: storage/aptosdb/src/db_options.rs (hunk 3 above)
Comment (grao1991) [reply storage/aptosdb/src/db_options.rs]: Offline discussed, split it into its own db as we don't see any downside here.

**Review by msmouse — APPROVED** [review]

File: config/src/config/storage_config.rs (hunk 1 above)
Comment (grao1991) [reply config/src/config/storage_config.rs]: Done

File: storage/aptosdb/src/ledger_db.rs (hunk 4 above)
Comment (grao1991) [reply storage/aptosdb/src/ledger_db.rs]: Done

**Review by areshand — APPROVED** [review]

**Pull request merged**